### DISCUSSION TITLE
Any provision on Stackstorm UI for reviewing the logs for past days or months and an option to export the log files?

### DISCUSSION THREAD

**Discussion by zsmanjot** [StackStorm/st2#5974] [UI]
In stackstorm UI, we could see the actions being taken on runtime and some older actions. But what if we want to have an option to view the logs for past days and months and wants to download a log file?

Is there any option on Stackstorm UI for the same , though I could not find one. Or if there is any setting that we need to enable in config files to get this done?

**Comment by zsmanjot** [comment]
Team,

Are there any updates on it?

**Comment by nzlosh** [comment]
The history tab is for this purpose.  If this history is too short you can adjust the data rention of the garbage collector https://docs.stackstorm.com/troubleshooting/purging_old_data.html#purging-old-operational-data.  Application logs are expected to managed by the system administrator, this could be in the form of flat files in `/var/log/st2`, in journald or sent to a central logging system.  As such, log file access is outside the scope of the webui.

**Reply by zsmanjot** [reply]
Thanks nzlosh,
Yes , I understand the history tab is for that pupose but it is realtime.  In an environment where stackstorm is extensively used and handles N number of alerts, it is not easy to read them out for even past one hour even after adjusting the data retention of the garbage collector.

Though we are storing the logs locally too , but this option to have on UI  will be hassle free and easily accessible . Means good to have an option  if we want to review data for certain timeperiod.

So just a suggestion if it can be considered,  the UI should give option to download the reports, atleast for past months or some weeks/days.
Let me know your views on it.

**Reply by nzlosh** [reply]
Have you tried using the st2 cli or rest API to fetch executions.  This might be better for suited high volumes of executions.  You can control which executions are collected by date and limit how many results are returned.

```
st2 execution list --help

  -s SORT_ORDER, --sort SORT_ORDER
                        Sort action executions by start timestamp,
                        asc|ascending (earliest first) or desc|descending
                        (latest first)
  -tg TIMESTAMP_GT, --timestamp-gt TIMESTAMP_GT
                        Only return executions with timestamp greater than the
                        one provided. Use time in the format
                        "2000-01-01T12:00:00.000Z".
  -tl TIMESTAMP_LT, --timestamp-lt TIMESTAMP_LT
                        Only return executions with timestamp lower than the
                        one provided. Use time in the format
                        "2000-01-01T12:00:00.000Z".
  -l, --showall
  -a ATTR [ATTR ...], --attr ATTR [ATTR ...]
                        List of attributes to include in the output. "all"
                        will return all attributes.
  -w WIDTH [WIDTH ...], --width WIDTH [WIDTH ...]
                        Set the width of columns in output.

  --action ACTION       Action reference to filter the list.
  --status STATUS       Only return executions with the provided status.
                        Possible values are 'succeeded', 'running',
                        'scheduled', 'paused', 'failed', 'canceling' or
                        'canceled'.
  --user USER           Only return executions created by the provided user.
  --trigger_instance TRIGGER_INSTANCE
                        Trigger instance id to filter the list.
```

**Comment by zsmanjot** [comment]
Yes I have even tried this option as well. It is ok to list them on Command line at run time and redirect into a file or so. But my ask was different . Like the way we have logs for Pagerduty , for nagios in their UIs . One click, select date and time and you can download your CSVs having the data. In similar fashion , what all actions ST2 took at what time, which workflow was run then filtering everything out, it can be fruitful.

**Comment by nzlosh** [comment]
OK, I understand, you'd like to get a set of actions in a summarised form, not actual st2 application logs, which sounds useful. You could create an issue in the st2web github repository explaining the feature.

**Reply by zsmanjot** [reply]
Sure, definitely. Let me do so.
